src/AsmResolver/Shims/RuntimeInformationShim.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace AsmResolver.Shims;
+
+/// <summary>
+/// Provides compatibility shims for the RuntimeInformation class for builds that target older .NET framework versions.
+/// </summary>
+public static class RuntimeInformationShim
+{
+    static RuntimeInformationShim()
+    {
+#if !NET35
+        IsRunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        IsRunningOnUnix = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        FrameworkDescription = RuntimeInformation.FrameworkDescription;
+#else
+        // https://stackoverflow.com/questions/5116977/how-to-check-the-os-version-at-runtime-e-g-on-windows-or-linux-without-using
+        int p = (int) Environment.OSVersion.Platform;
+        IsRunningOnUnix = p is 4 or 6 or 128;
+
+        IsRunningOnWindows = !IsRunningOnUnix;
+        FrameworkDescription = $".NET Framework {Environment.Version}";
+#endif
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether the current runtime runs on the Microsoft Windows operating system.
+    /// </summary>
+    public static bool IsRunningOnWindows
+    {
+        get;
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether the current runtime runs on a Unix-based operating system.
+    /// </summary>
+    public static bool IsRunningOnUnix
+    {
+        get;
+    }
```

Comment:
I think it's more efficient to not cache the value because then the JIT can optimize branching based on runtime information.

Reply:
As long as the static constructor has run when the JIT compiles a method using it, it will see this as a constant (because it gets emitted as a `static readonly` field, through a very small method body).

Reply:
I'm pretty sure it will also depend on whether the calls to into `RuntimeInformation` are considered JIT intrinsics, which I am not sure about.